src/auditwheel/patchelf.rs

```diff
@@ -1,97 +1,177 @@
 use anyhow::{bail, Context, Result};
+use arwen::elf::ElfContainer;
+use std::collections::HashMap;
 use std::ffi::OsStr;
-use std::path::Path;
-use std::process::Command;
-
-static MISSING_PATCHELF_ERROR: &str = "Failed to execute 'patchelf', did you install it? Hint: Try `pip install maturin[patchelf]` (or just `pip install patchelf`)";
-
-/// Verify patchelf version
-pub fn verify_patchelf() -> Result<()> {
-    let output = Command::new("patchelf")
-        .arg("--version")
-        .output()
-        .context(MISSING_PATCHELF_ERROR)?;
-    let version = String::from_utf8(output.stdout)
-        .context("Failed to parse patchelf version")?
-        .trim()
-        .to_string();
-    let version = version.strip_prefix("patchelf").unwrap_or(&version).trim();
-    let semver = version.parse::<semver::Version>().context(
-        "Failed to parse patchelf version, auditwheel repair requires patchelf >= 0.14.0.",
-    )?;
-    if semver < semver::Version::new(0, 14, 0) {
-        bail!(
-            "patchelf {} found. auditwheel repair requires patchelf >= 0.14.0.",
-            version
-        );
+use std::path::{Path, PathBuf};
+
+/// A struct for efficiently patching ELF files with multiple operations
+/// while avoiding re-parsing the ELF file multiple times.
+pub struct ElfPatcher {
+    file_path: PathBuf,
+    file_data: Vec<u8>,
+}
```

Comment:
```suggestion
pub struct ElfPatcher {
    file_path: PathBuf,
    elf: ElfContainer,
}
```

please refactor to this, you are still calling `ElfContainer::parse` in every methods, we just need to call it once in `ElfPatcher::new`.